src/lib/cimbar_js/cimbar_js.cpp

```diff
@@ -100,22 +101,37 @@ int encode(uint8_t* buffer, size_t size)
 	return 1;
 }
 
-int configure(unsigned color_bits)
+int configure(unsigned color_bits, unsigned ecc, int compression)
 {
-	if (color_bits != _colorBits and color_bits <= 3)
+	// defaults
+	if (color_bits > 3)
+		color_bits = 2;
+	if (ecc < 0 or ecc >= 150)
+		ecc = 30;
+	if (compression < 0 or compression > 22)
+		compression = 6;
+
+	// check if we need to refresh the stream
+	bool refresh = (color_bits != _colorBits or ecc != _ecc or compression != _compressionLevel);
+	if (refresh)
 	{
+		// update config
 		_colorBits = color_bits;
+		_ecc = ecc;
+		_compressionLevel = compression;
 
+		// try to refresh the stream
 		if (_window and _fes)
 		{
 			unsigned buff_size_new = cimbar::Config::fountain_chunk_size(_ecc, cimbar::Config::symbol_bits() + _colorBits);
-			if (!_fes->reset_and_resize_buffer(buff_size_new))
+			if (!_fes->restart_and_resize_buffer(buff_size_new))
```

Comment:
Decided to rename this -- it's usually used as a smart ptr, and `reset()` already means something.